### ISSUE TITLE
Setting up repo with yarn 3 fails

### ISSUE THREAD

**Issue by ferdnyc** [jupyter-widgets/ipywidgets#3797]
## Description

The build setup claims that it wants to be configured with yarn 3, so I attempted to do that. I was not able to complete the `yarn install` process successfully due to dependency conflicts.

## Reproduce

1. Check out the repo
2. Ensure that yarn 3 is used for repo builds (even if your system is otherwise running yarn 1) by running the command `yarn set version berry`. This will download the latest 3.x version and install it locally so that it becomes the default `yarn` used for the repo.
3. Attempt a `yarn install`. This will produce several resolution failures, due to undeclared dependencies. On my system:

> ```console
> $ yarn install
> ➤ YN0000: ┌ Resolution step
> ➤ YN0060: │ @jupyter-widgets/base-manager@workspace:packages/base-manager provides webpack (p4adfb) with version 5.82.1, which doesn't satisfy what istanbul-instrumenter-loader requests
> ➤ YN0060: │ @jupyter-widgets/base@workspace:packages/base provides webpack (pdd4b0) with version 5.82.1, which doesn't satisfy what istanbul-instrumenter-loader requests
> ➤ YN0002: │ @jupyter-widgets/controls@workspace:packages/controls doesn't provide caniuse-lite (p2e9eb), requested by postcss-cssnext
> ➤ YN0060: │ @jupyter-widgets/controls@workspace:packages/controls provides webpack (p47993) with version 5.82.1, which doesn't satisfy what istanbul-instrumenter-loader requests
> ➤ YN0002: │ @jupyter-widgets/html-manager@workspace:packages/html-manager doesn't provide caniuse-lite (p1db6d), requested by postcss-cssnext
> ➤ YN0002: │ @jupyter-widgets/jupyterlab-manager@workspace:python/jupyterlab_widgets doesn't provide react (p0e3ec), requested by @jupyterlab/settingregistry
> ➤ YN0002: │ @jupyter-widgets/jupyterlab-manager@workspace:python/jupyterlab_widgets doesn't provide webpack (p8f14c), requested by source-map-loader
> ➤ YN0002: │ @jupyterlab/services@npm:7.0.0 doesn't provide react (p7bfc4), requested by @jupyterlab/settingregistry
> ➤ YN0002: │ @lerna/version@npm:5.6.2 doesn't provide nx (p20b2d), requested by @nrwl/devkit
> ➤ YN0000: │ Some peer dependencies are incorrectly met; run yarn explain peer-requirements <hash> for details, where <hash> is the six-letter p-prefixed code
> ➤ YN0000: └ Completed in 0s 582ms
> ➤ YN0000: ┌ Fetch step
> ```

4. The documentation states that undeclared dependencies can be fixed locally via the [`packageExtensions` directive of the `.yarnrc.yml` file](https://yarnpkg.com/getting-started/migration#fix-dependencies-with-packageextensions). I ended up with the following complete contents of `.yarnrc.yml` (including automatically inserted lines):

```yml
nodeLinker: node-modules

yarnPath: .yarn/releases/yarn-3.6.1.cjs

packageExtensions:
  "@nrwl/devkit@*":
    dependencies:
      nx: "^15"
  "@jupyterlab/settingregistry@*":
    dependencies:
      react: "^18"
  "source-map-loader@*":
    dependencies:
      webpack: "^5"
  "postcss-cssnext@*":
    dependencies:
      caniuse-lite: "^1"
```

5. Run `yarn install` again. Everything is solved except for `istanbul-instrumenter-loader`'s incompatibility with webpack 5:

```console
$ yarn install
➤ YN0000: ┌ Resolution step
➤ YN0060: │ @jupyter-widgets/base-manager@workspace:packages/base-manager provides webpack (p4adfb) with version 5.82.1, which doesn't satisfy what istanbul-instrumenter-loader requests
➤ YN0060: │ @jupyter-widgets/base@workspace:packages/base provides webpack (pdd4b0) with version 5.82.1, which doesn't satisfy what istanbul-instrumenter-loader requests
➤ YN0060: │ @jupyter-widgets/controls@workspace:packages/controls provides webpack (p47993) with version 5.82.1, which doesn't satisfy what istanbul-instrumenter-loader requests
➤ YN0000: │ Some peer dependencies are incorrectly met; run yarn explain peer-requirements <hash> for details, where <hash> is the six-letter p-prefixed code
➤ YN0000: └ Completed in 1s 981ms
➤ YN0000: ┌ Fetch step
```

...And that's as far as I could get. [webpack-contrib/istanbul-instrumenter-loader](https://github.com/webpack-contrib/istanbul-instrumenter-loader) is an archived project as of January 2021, it has apparently been abandoned and will not be updated. Its `peerDependencies` are declared as: `"webpack": "^2.0.0 || ^3.0.0 || ^4.0.0"`, meaning that webpack 5 will not satisfy it. But it's (still!) a `devDependency` of even the latest version of several jupyter-widgets packages. 

In fact, in the case of `@jupyter-widgets/controls`, the `devDependencies` are internally inconsistent, AFAICT, since it requires `webpack@^5.65.0` along with `istanbul-instrumenter-loader@^3.0.1` which rejects webpack 5:

```console
$ npm info "@jupyter-widgets/controls" devDependencies
[...]
{
  '@jupyterlab/services': '^6.0.0',
  '@types/d3-color': '^3.0.2',
  '@types/d3-format': '^3.0.1',
  '@types/expect.js': '^0.3.29',
  '@types/mathjax': '^0.0.37',
  '@types/mocha': '^9.0.0',
  '@types/node': '^17.0.2',
  chai: '^4.0.0',
  'css-loader': '^6.5.1',
  'expect.js': '^0.3.1',
  'istanbul-instrumenter-loader': '^3.0.1',
  karma: '^6.3.3',
  'karma-chrome-launcher': '^3.1.0',
  'karma-coverage': '^2.0.3',
  'karma-firefox-launcher': '^2.1.1',
  'karma-ie-launcher': '^1.0.0',
  'karma-mocha': '^2.0.1',
  'karma-mocha-reporter': '^2.2.5',
  'karma-webpack': '^5.0.0',
  less: '^4.1.2',
  mocha: '^9.0.0',
  'npm-run-all': '^4.1.5',
  postcss: '^8.3.2',
  'postcss-cli': '^9.1.0',
  'postcss-cssnext': '^3.1.0',
  'postcss-import': '^14.0.2',
  'postcss-loader': '^6.1.0',
  rimraf: '^3.0.2',
  sinon: '^12.0.1',
  'sinon-chai': '^3.3.0',
  'style-loader': '^3.3.1',
  typescript: '~4.9.4',
  webpack: '^5.65.0'
}
```

Unless I'm missing something, this makes it impossible to successfully resolve the dependencies within yarn 3, as it seemingly considers inconsistent dependencies to be a fatal error. Dependencies cannot (according to the same docs) be _modified_ using `packageExtensions`, only added.

So that's where I'm stuck. I have no idea how to make the dependencies resolvable in a way that will satisfy yarn 3. I tried forcing resolution to webpack 4, but that only caused much bigger conflicts with even more packages.

## Expected behavior

Yarn installs the necessary packages.

## Context

- ipywidgets version repo `main` branch checkout
- Operating System and version: Fedora Linux 38 x86_64
- Browser and version: N/a
- yarn version 3.6.1

**Comment by ferdnyc** [comment]
It's also possible to add undeclared dependencies as `peerDependencies` of the package failing to fulfill the request, instead of as `dependencies` of the one making it. That route requires much more extensive changes; for example, moving the `react` dependency of `@jupyterlab/settingsregistry` and the `webpack` dependency of `source-map-loader` to instead be `peerDependencies` of `@jupyter-widgets/jupyterlab-manager` required all of the following additional changes, making me conclude it was the wrong approach:

```yml
  "@jupyter-widgets/jupyterlab-manager@*":
    peerDependencies:
      webpack: "^5"
      react: "^18"
  "@jupyter-widgets/base-manager@*":
    peerDependencies:
      react: "^18"
  "@jupyter-widgets/base@*":
    peerDependencies:
      react: "^18"
  "@jupyterlab/services@*":
    peerDependencies:
      react: "^18"
  "@jupyterlab/logconsole@*":
    peerDependencies:
      react: "^18"
  "@jupyterlab/docregistry@*":
    peerDependencies:
      react: "^18"
  "@jupyterlab/application@*":
    peerDependencies:
      react: "^18"
  "@jupyter-widgets/notebook-manager@*":
    peerDependencies:
      react: "^18"
  "@jupyterlab/rendermime@*":
    peerDependencies:
      react: "^18"
  "@jupyterlab/outputarea@*":
    peerDependencies:
      react: "^18"
  "@jupyterlab/translation@*":
    peerDependencies:
      react: "^18"
  "@jupyter-widgets/html-manager@*":
    peerDependencies:
      react: "^18"
  "@jupyter-widgets/example-web1@*":
    peerDependencies:
      react: "^18"
  "@jupyter-widgets/example-web2@*":
    peerDependencies:
      react: "^18"
  "@jupyter-widgets/example-web3@*":
    peerDependencies:
      react: "^18"
  "@jupyter-widgets/example-web4@*":
    peerDependencies:
      react: "^18"
  "@jupyter-widgets/controls@*":
    peerDependencies:
      react: "^18"
  "@jupyter-widgets/output@*":
    peerDependencies:
      react: "^18"
  "@jupyterlab/lsp@*":
    peerDependencies:
      react: "^18"
```

This had no impact on the istanbul-instrumenter-loader dependency failures... except to cause all of them to be listed _twice_ in the `yarn install` output, for some reason:

```console
$ yarn install
➤ YN0000: ┌ Resolution step
➤ YN0060: │ @jupyter-widgets/base-manager@workspace:packages/base-manager provides webpack (p5fef9) with version 5.82.1, which doesn't satisfy what istanbul-instrumenter-loader requests
➤ YN0060: │ @jupyter-widgets/base-manager@workspace:packages/base-manager [a3660] provides webpack (p74dad) with version 5.82.1, which doesn't satisfy what istanbul-instrumenter-loader requests
➤ YN0060: │ @jupyter-widgets/base@workspace:packages/base provides webpack (p7e39c) with version 5.82.1, which doesn't satisfy what istanbul-instrumenter-loader requests
➤ YN0060: │ @jupyter-widgets/base@workspace:packages/base [a3660] provides webpack (p08a4f) with version 5.82.1, which doesn't satisfy what istanbul-instrumenter-loader requests
➤ YN0060: │ @jupyter-widgets/controls@workspace:packages/controls provides webpack (p82808) with version 5.82.1, which doesn't satisfy what istanbul-instrumenter-loader requests
➤ YN0060: │ @jupyter-widgets/controls@workspace:packages/controls [a3660] provides webpack (p93aa1) with version 5.82.1, which doesn't satisfy what istanbul-instrumenter-loader requests
➤ YN0000: │ Some peer dependencies are incorrectly met; run yarn explain peer-requirements <hash> for details, where <hash> is the six-letter p-prefixed code
➤ YN0000: └ Completed in 0s 921ms
➤ YN0000: ┌ Fetch step
```

**Comment by ferdnyc** [comment]
So, two things I've discovered/realized:

1. Since the jupyter-widgets packages are all part of the repo, in `packages/`, it's a simple enough matter to just _remove_ the `istanbul-instrumenter-loader` dependency from `base/`, `base-manager/`, and `controls/` by simply editing the packages. So I removed the `devDependency` from each directory's `package.json'`, and edited the local `tests/webpack-cov.conf.js` to delete the installation of the loader. That made my `yarn install` Resolution step complete no messages whatsoever.
2. Those messages apparently aren't the fatal errors I believed them to be. The reason I _thought_ they were fatal errors, and the reason `yarn install` never completes for me, is because it's **segfaulting during the Fetch step**. Silently.

I was only able to work out the second point by running `strace -f yarn install`, at which point I can see the segfault occurring in one of the child processes. But this apparently goes unnoticed by yarn, which simply stops all activity with no messages displayed and an exit code of 0 (success).
